channel/funder.go

```diff
@@ -0,0 +1,231 @@
+// Copyright 2023 PolyCrypt GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package channel
+
+import (
+	"context"
+	"errors"
+	"github.com/stellar/go/keypair"
+	"github.com/stellar/go/xdr"
+	"log"
+	pchannel "perun.network/go-perun/channel"
+	"perun.network/perun-stellar-backend/channel/env"
+	"perun.network/perun-stellar-backend/wallet"
+	"perun.network/perun-stellar-backend/wire"
+	"time"
+)
+
+const MaxIterationsUntilAbort = 20
+const DefaultPollingInterval = time.Duration(6) * time.Second
+
+type Funder struct {
+	stellarClient   *env.StellarClient
+	acc             *wallet.Account
```

Comment:
This is unused. Also I don't see a reason why the funder should have access to the channel participant's secret key